contracts/deployless/EstimationStructs.sol

```diff
@@ -18,15 +21,22 @@ contract EstimationStructs {
 
   // We do not care about nonces here, unlike portfolio simulations
   // In portfolio simulations we may want to simulate multiple AccountOps and see what nonce we started with, to know where the executing node is
-  // Here, we only care about one particular AccountOp (and potentially accountOpToExecuteBefore for recovery finalization)
+  // Here, we only care about one particular AccountOp
   struct SimulationOutcome {
     uint gasUsed;
     bool success;
     bytes err;
+    bool initialGasLimitFailed;
   }
 
   struct FeeTokenOutcome {
     uint gasUsed;
     uint amount;
   }
+
+  struct GasLimits {
+    uint256 gasLimit;
+    uint256 upperBoundLimit;
+    bool shouldRevertIfConditionsMet;
```

Comment:
maybe rename this to `shouldRevertUponSuccess` if there are no other conditions to this, and add a comment explaining it
// whether we should revert with the outcome of a successful simulation; the goal of this is to simulate an accountOp WITHOUT modifying the state, so we can simulate it again later